### PR TITLE
Allow subqueries without aliases

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/2417

 # Rationale for this change

I am trying to run some ANSI SQL that works fine in Apache Spark but fails in DataFusion because we currently require subqueries to have aliases.

As described in the [issue](https://github.com/apache/arrow-datafusion/issues/2417) though, Postgres *does* require subqueries to be aliased, so that is probably what determined the current behavior?

Should we make this behavior configurable?

There was a discussion about this in https://github.com/apache/arrow-datafusion/pull/1067 

# What changes are included in this PR?

Remove the check that caused subqueries without aliases to fail. Note that there were no tests for this functionality.

# Are there any user-facing changes?

No

### CODE REVIEW

**Comment by andygrove** [comment]
@alamb @yjshen @xudong963 @houqp How do you think we should handle this? Should I consider enabling this or not based on the SQL dialect being used?

File: datafusion/core/tests/sql/select.rs [hunk 1]
```diff
@@ -925,6 +925,38 @@ async fn csv_select_nested() -> Result<()> {
...
+        "| a  | 5  | 65   |",
+        "| a  | 6  | -101 |",
+        "| a  | 6  | -31  |",
+        "| a  | 6  | 36   |",
+        "+----+----+------+",
+    ];
+    assert_batches_eq!(expected, &actual);
+    Ok(())
+}
+
```
Comment (alamb) [inline datafusion/core/tests/sql/select.rs:959]: Maybe we could also add a test here that had two unaliased subqueries (should result in a cross join) as a potential other case that would be good to cover

**Review by alamb — APPROVED** [review]
I don't honestly know the deeper background and understanding of why postgres requires an alias or if there are deeper implications to allowing unaliased subqueries.

However, I think the additional test coverage is 👍  And the fact there are no additional errors is a good sign to me.

File: datafusion/core/tests/sql/select.rs (hunk 1 above)
Comment (andygrove) [reply datafusion/core/tests/sql/select.rs:959]: Thanks for the review @alamb. I added the suggested test and did not run into problems so will go ahead and merge this one.

Comment (alamb) [reply datafusion/core/tests/sql/select.rs:959]: Awesome!

**Pull request merged**